src/mongocxx/config/compiler.hpp

```diff
@@ -0,0 +1,15 @@
+// Copyright 2015 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#define MONGOCXX_INLINE inline __attribute__((__visibility__("hidden"), __always_inline__))
```

Comment:
Hmm... we could only use BSONCXX_INLINE? or we could call the bsoncxx prelude/postlude from the mognocxx prelude/postlude.

Or we could push/pop all the bsoncxx complier headers in the the mongocxx prelude/postlude and change this file to just:
```
#define MONGOCXX_INLINE BSONCXX_INLINE
```